libs/sysaudio/src/coreaudio.zig

```diff
@@ -138,28 +138,31 @@ pub const Context = struct {
                 }
             }
 
-            prop_address.mSelector = if (is_darling) c.kAudioDevicePropertyPreferredChannelsForStereo else c.kAudioDevicePropertyPreferredChannelLayout;
-            prop_address.mScope = if (mode == .playback) c.kAudioObjectPropertyScopeOutput else c.kAudioObjectPropertyScopeInput;
-            if (c.AudioObjectGetPropertyDataSize(id, &prop_address, 0, null, &io_size) != 0) {
+            // for now, only set `channels` to the output channels
+            const audio_buffer_list_property_address = c.AudioObjectPropertyAddress{
+                .mSelector = c.kAudioDevicePropertyStreamConfiguration,
+                .mScope = c.kAudioDevicePropertyScopeOutput,
+                .mElement = c.kAudioObjectPropertyElementMain,
+            };
+            var output_audio_buffer_list: c.AudioBufferList = undefined;
+            var audio_buffer_list_size: c_uint = undefined;
+
+            if (c.AudioObjectGetPropertyDataSize(id, &audio_buffer_list_property_address, 0, null, &audio_buffer_list_size) != c.noErr) {
                 return error.OpeningDevice;
+                // return error.FailedToGetAudioBufferListSize;
             }
 
-            var channel_layout: c.AudioChannelLayout = undefined;
-            if (c.AudioObjectGetPropertyData(
-                id,
-                &prop_address,
-                0,
-                null,
-                &io_size,
-                &channel_layout,
-            ) != 0) {
+            if (c.AudioObjectGetPropertyData(id, &prop_address, 0, null, &audio_buffer_list_size, @ptrCast(*anyopaque, &output_audio_buffer_list)) != c.noErr) {
```

Comment:
`@ptrCast` can be omitted.
also can you change all error checks from to `!= 0` to `!= c.noErr`?

Reply:
Done